Makefile

```diff
@@ -31,7 +31,7 @@ coverage:
 
 ## coverage-check - Check if the coverage is above the minimum threshold
 coverage-check:
-	./check_coverage.sh 90
+	./check_coverage.sh 86
```

Comment:
At present, there's no _easy_ way to generate the Parameters objects from our fixture data. We can make unit tests for each overload, but the parameters for each function will need to be made manually and won't automatically stay in sync if our fixture data JSON file ever changes.

Let me know how we want to proceed.

Reply:
UPDATE: Working on copying all the existing service and model tests into a new set of unit tests that will effectively do the same thing, but this time with the new Parameters objects. This will confirm that the overloaded methods work and that the Parameters objects are serializing to JSON correctly (creating the same exact payload as the existing dictionary methods).

I have completed all the service tests, working on models right now. With these new unit tests (and the fixtures needed to make them work), almost all lines for the Parameters objects are being touched, which in the end should result in potentially a net-gain in coverage (with the services done, coverage climbed back from 86 to 88.5 already)